device.yml

```diff
@@ -0,0 +1,116 @@
+%YAML 1.1
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/harp-tech/reflex-generator/main/schema/device.json
+device: InputExpander
+whoAmI: 1106
+firmwareVersion: "2.2"
+hardwareTargets: "1.2"
+registers:
+  AuxInPort:
+    address: 32
+    type: U8
+    access: Event
+    maskType: AuxiliaryInputs
+    description: Reports the state of the auxiliary inputs.
+  AuxInRisingEdge:
+    address: 33
+    type: U8
+    access: Write
+    maskType: AuxiliaryInputs
+    description: Enables rising edge detection on the auxiliary inputs.
+  AuxInFallingEdge:
+    address: 34
+    type: U8
+    access: Write
+    maskType: AuxiliaryInputs
+    description: Enables falling edge detection on the auxiliary input port.
+  DigitalInput:
+    address: 35
+    type: U16
+    access: Event
+    length: 2
+    payloadSpec:
+      DigitalInputState:
+        offset: 0
+        maskType: DigitalInputs
+        description: Reports the state of all digital inputs in the port.
+      DigitalInputChanged:
+        offset: 1
+        maskType: DigitalInputs
+        description: Reports which digital inputs changed state in the port.
+    description: Reports the state of the digital inputs.
+  DigitalInputEnableRisingEdge:
+    address: 36
+    type: U16
+    access: Write
+    maskType: DigitalInputs
+    description: Enables rising edge detection on the digital input port.
+  DigitalInputFallingEdge:
+    address: 37
+    type: U16
+    access: Write
+    maskType: DigitalInputs
+    description: Enables falling edge detection on the digital input port.
+  InputSamplingMode:
+    address: 38
+    type: U8
+    access: Write
+    maskType: InputSamplingConfig
+    description: Configures the input sampling mode.
+  EncoderSamplingMode:
+    address: 39
+    type: U8
+    access: Write
+    maskType: EncoderSamplingConfig
+    description: Configures the rotary encoder sampling mode.
+  EncoderData:
+    address: 40
+    type: S16
+    access: Event
+    description: Reports the value of the current read from the rotary encoder.
+  ExpansionBoard:
+    address: 41
+    type: U8
+    access: Write
+    maskType: ExpansionBoardType
+    description: Selects the board to be interfaced with via the expansion port.
+bitMasks:
+  AuxiliaryInputs:
+    bits:
+      Aux0: 0x1
+      Aux1: 0x2
+      Aux0Changed: 0x20
+      Aux1Changed: 0x40
+    description: Specifies the state of auxiliary input lines.
+  DigitalInputs:
+    bits:
+      DI0: 0x1
+      DI1: 0x2
+      DI2: 0x4
+      DI3: 0x8
+      DI4: 0x10
+      DI5: 0x20
+      DI6: 0x40
+      DI7: 0x80
+      DI8: 0x100
+      DI9: 0x200
+    description: Specifies the state of the digital input lines.
+groupMasks:
+  InputSamplingConfig:
+    values:
+      OnInterrupt: 0
+      OnPooling1kHz: 1
+      OnPooling2Hz: 2
+    description: Specifies the input mode configuration from the available list of options.
+  EncoderSamplingConfig:
+    values:
+      Disabled: 0
+      Pooling250Hz: 1
```

Comment:
Is there a reason for using `OnPolling1kHz` in the above enum and `Polling1kHz` here? At first I thought maybe these don't refer to events, but then there is also `OnMovement` below, so I was confused.

In general for `Polling` I would drop the `On` prefix, and even in general I don't think we are using this prefix `On` for other events.

Reply:
Yep I agree with dropping the prefix from the Polling, but i think it makes sense to leave it on the OnMovement, since movement alone might not indicate the time during which the sample is taken.